mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java

```diff
@@ -1015,6 +1019,7 @@ public void run() {
         if (AppConstants.Versions.feature24Plus) {
             maybeShowSetDefaultBrowserDialog(sharedPreferences, appContext);
         }
+        mPreferenceManager.registerOnSharedPreferenceChangeListener(this);
```

Comment:
The listener never get removed, it leaks the BrowserApp activity

Reply:
done